Changing License

marcelofcastro opened the pull request:
This PR updates the license, copyright information and add the CLA assistant workflow. 
 - The license was update to BSD 3 and copyright information (in readme and in .mo file) was updated.
 - Contributors now have to read and virtually sign their PRs in order to allow merging.

lvanfretti commented:
Adding .pdf with emails that show the approval from RTE, AIA and DTU, which had the (c) of iPSL to make the license change of OpenIPSL requested
[License Agreement for OpenIPSL.pdf](https://github.com/OpenIPSL/OpenIPSL/files/7654222/License.Agreement.for.OpenIPSL.pdf)
.

dietmarw commented on OpenIPSL/Resources/legal/cla-entity.md:
```diff
@@ -0,0 +1,85 @@
+
+Thank you for your interest in contributing to Open Instance Power System Library - OpenIPSL  ("We" or "Us").
+
+This contributor agreement ("Agreement") documents the rights granted by contributors to Us. To make this document effective, please sign it and send it to Us by electronic submission, following the instructions [here](https://github.com/cla-assistant/github-action). This is a legally binding document, so please read it carefully before agreeing to it. The Agreement may cover more than one softwa ...
```
That link does not give very useful instruction to a normal end user other than a sysadmin that shall implement the CLA on a server. So the submission process should be described for normal human beings.

marcelofcastro replied:
True true, should we link it to a page that has the step-by-step procedure of working with a library in github or should we create our own page and place it in Resources? What do you think is best?

marcelofcastro replied:
Maybe this [link](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) you sent might help me. I will take a look at it first.

dietmarw replied:
That document would normally get presented to users providing a PR by the automated CLA process we want to implement. So they ARE already in the middle of accepting the CLA. All they would need as information is what the CLA means not how it is done since this is already triggered.

dietmarw commented on OpenIPSL/Resources/legal/cla-individual.md:
```diff
@@ -0,0 +1,91 @@
+
+Thank you for your interest in contributing to Open Instance Power System Library - OpenIPSL  ("We" or "Us").
+
+This contributor agreement ("Agreement") documents the rights granted by contributors to Us. To make this document effective, please sign it and send it to Us by electronic submission, following the instructions [here](https://github.com/cla-assistant/github-action). This is a legally binding document, so please read it carefully before agreeing to it. The Agreement may cover more than one softwa ...
```
Same problem with the instructions like mentioned before.

dietmarw commented on README.md:
```diff
@@ -30,16 +30,6 @@ We recommend you start from there.
-
 ## Citing OpenIPSL in Publications:
 If you use OpenIPSL in your work or research, all we ask you in exchange is that you **cite the reference publications**, according to your use.
 Please consult our [Publication List](http://openipsl.readthedocs.io/en/latest/publications.html), for browsing the reference publications.
```
Do NOT link to the readthedocs page. That one should actually be removed and never mentioned again since we do not keep that up to date.

dietmarw commented on README.md:
```diff
@@ -48,29 +38,28 @@ Preferably, please cite the following paper:
 
-The OpenIPSL is released under the same license as former iPSL project.
-All changes applied in this fork have the following copyright statement.
+Please consult the [contribution guidelines](http://openipsl.readthedocs.io/en/latest/community/contributing.html). More information about our contributors can be found over [here](http://openipsl.readthedocs.io/en/latest/community/community-home.html). 
```
No link to readthedocs. Better create a new file following: https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors and link to that one.

dietmarw commented on README.md:
```diff
@@ -48,29 +38,28 @@ Preferably, please cite the following paper:
+## Copyright and License Information:
 
+### OpenIPSL:
+Copyright April 2016 - current [Luigi Vanfretti](https://github.com/lvanfretti), [ALSETLab](https://github.com/ALSETLab), Troy, NY (Formely [SmarTS Lab](http://openipsl.readthedocs.io/en/latest/community/community-home.html), Stockholm, Sweden). 
```
no readthedocs use again the functionality of documents as described in https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

dietmarw commented on README.md:
```diff
@@ -48,29 +38,28 @@ Preferably, please cite the following paper:
 This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
-If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0.
+If a copy of the MPL was not distributed with this file, you can obtain one at https://www.mozilla.org/en-US/MPL/2.0/.
 
```
Now that there is only ONE license (thanks, there is no need to quote the old versions etc).

marcelofcastro replied:
Wait, so should we remove the whole iPSL part? Or just the part that says that iPSL is licensed under MPL v2? I didn't quite get it.

dietmarw replied:
**All** authors have agreed to the license change. This includes especially the parts which stem from the iPSL. So there is no longer any need to split this up. All the code is now under BSD.

marcelofcastro replied:
okdok

marcelofcastro replied:
then we should also remove the ipsl part from the Copyright.mo, correct?

lvanfretti commented:
Yes! 

> On Mar 3, 2022, at 19:33, Marcelo de C. Fernandes ***@***.***> wrote:
> 
> ﻿
> @marcelofcastro commented on this pull request.
> 
> In README.md:
> 
> >  The authors can be contacted by email: ***@***.***
>  This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
> -If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0.
> +If a copy of the MPL was not distributed with this file, you can obtain one at https://www.mozilla.org/en-US/MPL/2.0/.
>  
> then we should also remove the ipsl part from the Copyright.mo, correct?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.

marcelofcastro commented:
I have added a contribution guide together with a code of conduct for future developers. I have also added a file that has some information about CLA and how to sign it. I've edited the README file to incorporate these changes and to remove all references to read-the-docs.

I also believe that the instructions created (especially if we consider one of the Note: observations I made in the contribution guidelines) also address the Issue #201.

Furthermore, I have also asked people to use our Discussions page instead of creating an Issue for minor things. If it is necessary, the discussion can become an Issue and I believe we will have a much better interaction with users that way, while avoiding unnecessary issues to be open.

lvanfretti commented:
Marcelo, everything looks good to me, thanks so much for all this efforts!
Regarding to #201, your proposal makes things clearer, so we should go for it.

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+
+Thank you for considering investing your time to contribute to OpenIPSL! :tada::+1:
+
+Here, you will find a set of guidelines for contributing to the OpenIPSL, which is hosted in the [OpenIPSL page](https://github.com/OpenIPSL) on GitHub. Many of the instructions in this document are considered to be suggestions rather than strict rules, so please use your best judgement when reading this document and when contributing to OpenIPSL. In addition, you can find some tips and good prac ...
```
```suggestion
Here, you will find a set of guidelines for contributing to the OpenIPSL, which is hosted on the [OpenIPSL page](https://github.com/OpenIPSL/OpenIPSL) on GitHub. Many of the instructions in this document are considered to be suggestions rather than strict rules, so please use your best judgement when reading this document and when contributing to OpenIPSL. In addition, you can find some tips and good practices in [GitHub's Contribution Guideline](https://github.com/github/docs/blob/3c5055fc4eecf3f83560419d94be54458c2f56a1/CONTRIBUTING.md) and in [Atom's Contribution Guideline](https://github.com/atom/atom/blob/master/CONTRIBUTING.md).
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+
+### Join GitHub
+
+However you decide to contribute to OpenIPSL you should fist create your GitHub account, if you do not have one already. To create your account, just go to this [website](https://github.com/join), fill out the form, complete the captcha and click on Create Account. After that, you will need some material to get started with your git skills. 
```
```suggestion
However you decide to contribute to OpenIPSL you should fist create your GitHub account, if you do not have one already. To create your account, just go to [GitHub's join page](https://github.com/join), fill out the form, complete the captcha and click on Create Account. After that, you will need some material to get started with your git skills. 
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+Git is a version control solution that is used to manage the library and its model packages. The following is a list of things that can help you with getting started with Git.
+
+
+* Since you already have a GitHub account at this point, you can start by setting up Git. Useful information about that is found [here](https://docs.github.com/en/get-started/quickstart/set-up-git), but you can also use your web search tool to look for other sources of information.
```
```suggestion
* Since you already have a GitHub account at this point, you can start by setting up Git. Useful information about that is found on [GitHub's quick start guide](https://docs.github.com/en/get-started/quickstart/set-up-git), but you can also use your web search tool to look for other sources of information.
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+
+* Since you already have a GitHub account at this point, you can start by setting up Git. Useful information about that is found [here](https://docs.github.com/en/get-started/quickstart/set-up-git), but you can also use your web search tool to look for other sources of information.
+
+* [Basic information](https://www.freecodecamp.org/news/what-is-git-and-how-to-use-it-c341b049ae61/) and [courses](https://www.coursera.org/learn/introduction-git-github?specialization=google-it-automation&utm_source=gg&utm_medium=sem&utm_campaign=11-GoogleITwithPython-US&utm_content=B2C&campaignid=8986236679&adgroupid=119480419197&device=c&keyword=&matchtype=&network=g&devicemodel=&adpostion=&cr ...
```
```suggestion
* [Basic information](https://www.freecodecamp.org/news/what-is-git-and-how-to-use-it-c341b049ae61/) and [courses](https://www.coursera.org/learn/introduction-git-github) about Git are available in the internet and we strongly encourage you to look for those materials if you are not used to version control and Git.
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+Once you have created your GitHub account and you familiarized yourself with Git and GitHub, it is time for you to get to know the typical preferred workflow for contributing to the OpenIPSL. This will allow your contribution to be reviewed by the OpenIPSL developers before it is integrated into the library. The following is a list showing the steps that should be taken in order for a contributio ...
+
+
+* **Fork the OpenIPSL.** A fork is a copy of a repository that will be hosted, in this case, in your GitHub account. More information about forks is found [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
```
```suggestion
* **Fork the OpenIPSL.** A fork is a copy of a repository that will be hosted, in this case, in your GitHub account. More information about forks is found on [GitHub's quick start guide](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+
+
+* **Fork the OpenIPSL.** A fork is a copy of a repository that will be hosted, in this case, in your GitHub account. More information about forks is found [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+* **Create a Branch.** The branch will allow you to work with a version of your fork. It also allows for other people to contribute with your work without compromising the main branch of the fork and, especially, the main branch of the OpenIPSL. A branch name usually describes the main purpose of that branch. More information about branches is found [here](https://docs.github.com/en/get-started/q ...
```
```suggestion
* **Create a Branch.** The branch will allow you to work with a version of your fork. It also allows for other people to contribute with your work without compromising the main branch of the fork and, especially, the main branch of the OpenIPSL. A branch name usually describes the main purpose of that branch. More information about branches is found on [GitHub's quick start guide](https://docs.github.com/en/get-started/quickstart/github-flow).
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+
+* **Fork the OpenIPSL.** A fork is a copy of a repository that will be hosted, in this case, in your GitHub account. More information about forks is found [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+* **Create a Branch.** The branch will allow you to work with a version of your fork. It also allows for other people to contribute with your work without compromising the main branch of the fork and, especially, the main branch of the OpenIPSL. A branch name usually describes the main purpose of that branch. More information about branches is found [here](https://docs.github.com/en/get-started/q ...
+* **Make Changes to the Branch.** You can work on your branch by creating, deleting and modifying branches. To your changes to take effect, you will need to stage the altered files and to commit them. Please use meaningful messages when commiting your changes. The modifications you perform will be available in your fork's branch once you push your local copy to its remote version. More informatio ...
```
```suggestion
* **Make Changes to the Branch.** You can work on your branch by creating, deleting and modifying branches. To your changes to take effect, you will need to stage the altered files and to commit them. Please use meaningful messages when commiting your changes. The modifications you perform will be available in your fork's branch once you push your local copy to its remote version. More information about this procedure can be consulted on [GitHub's get started guide](https://docs.github.com/en/get-started/using-git/about-git).
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+* **Fork the OpenIPSL.** A fork is a copy of a repository that will be hosted, in this case, in your GitHub account. More information about forks is found [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+* **Create a Branch.** The branch will allow you to work with a version of your fork. It also allows for other people to contribute with your work without compromising the main branch of the fork and, especially, the main branch of the OpenIPSL. A branch name usually describes the main purpose of that branch. More information about branches is found [here](https://docs.github.com/en/get-started/q ...
+* **Make Changes to the Branch.** You can work on your branch by creating, deleting and modifying branches. To your changes to take effect, you will need to stage the altered files and to commit them. Please use meaningful messages when commiting your changes. The modifications you perform will be available in your fork's branch once you push your local copy to its remote version. More informatio ...
+* **Create the Pull Request.** After you are done with your changes, it is possible to submit a Pull Request (PR), which means that you will formally submit changes to be incorporated into OpenIPSL. Information about collaboration through PRs can be found [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests).
```
```suggestion
* **Create the Pull Request.** After you are done with your changes, it is possible to submit a Pull Request (PR), which means that you will formally submit changes to be incorporated into OpenIPSL. Information about collaboration through PRs can be found on [GitHub's documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests).
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+* **Create a Branch.** The branch will allow you to work with a version of your fork. It also allows for other people to contribute with your work without compromising the main branch of the fork and, especially, the main branch of the OpenIPSL. A branch name usually describes the main purpose of that branch. More information about branches is found [here](https://docs.github.com/en/get-started/q ...
+* **Make Changes to the Branch.** You can work on your branch by creating, deleting and modifying branches. To your changes to take effect, you will need to stage the altered files and to commit them. Please use meaningful messages when commiting your changes. The modifications you perform will be available in your fork's branch once you push your local copy to its remote version. More informatio ...
+* **Create the Pull Request.** After you are done with your changes, it is possible to submit a Pull Request (PR), which means that you will formally submit changes to be incorporated into OpenIPSL. Information about collaboration through PRs can be found [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests).
+* **Address Reviewer's Comments.** The procedure for having your contributions added to the OpenIPSL via PR is an iterative process. This means that you will submit your suggested changes and it is very likely that we will review your code and request for changes to be implemented. This back-and-forth part will lead to discussions and the main benefit from this process is the possibility of makin ...
```
```suggestion
* **Address Reviewer's Comments.** The procedure for having your contributions added to the OpenIPSL via PR is an iterative process. This means that you will submit your suggested changes and it is very likely that we will review your code and request for changes to be implemented. This back-and-forth part will lead to discussions and the main benefit from this process is the possibility of making our library better and even more robust. You can apply suggested changes directly through the GitHub UI. You can make any other changes in your fork, then commit them to your branch. When you push your branch, the modifications will automatically appear in the open PR.
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+### Improving Documentation :memo:
+
+The documentation of the OpenIPSL is located within the Modelica package. This documentation is added via annotations in the Modelica models and there are even dedicated files inside the library that are basically composed of annotations.
+In addition to that, markdown files are present in many folders of the library, providing description to the many aspects of this library.
```
```suggestion
```

dietmarw commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,122 @@
+The documentation of the OpenIPSL is located within the Modelica package. This documentation is added via annotations in the Modelica models and there are even dedicated files inside the library that are basically composed of annotations.
+In addition to that, markdown files are present in many folders of the library, providing description to the many aspects of this library.
+
+Indeed, the main goal of the documentation is to provide information on the development conventions adopted for this project, and provide a guide to the contributors of the project. If you want to contribute to building a better documentation, you can first start by learning about [Markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-gi ...
```
```suggestion
Indeed, the main goal of the documentation is to provide information on the development conventions adopted for this project, and provide a guide to the contributors of the project. If you want to contribute to building a better documentation, you can start by learning about [Modelica Annotations](https://mbe.modelica.university/behavior/equations/annotations/) if you are not familiar with this concept yet. Please, visit the [Issues List](https://github.com/OpenIPSL/OpenIPSL/issues) since there are Documentation-related issues that can help you better understand what is needed by OpenIPSL.
```

dietmarw commented on .github/legal/README.md:
```diff
@@ -0,0 +1,26 @@
+
+## 3. How can I sign the CLA?
+
+Contributors can sign the CLA within the PR by copying the statement **"I have read the CLA Document and I hereby sign the CLA"** and pasting it in the PR as a comment. This means that you agree with all terms that are described in the individual/entity CLA, whichever applies to your case as a contributor. <br/> 
```
But is this manual process not obsoleted by the automatic CLA signing process as part of the PR. This paragraph only causes confusion so I would remove it.

marcelofcastro replied:
Well, the automatic CLA asks for the user to copy and paste the statement, which is his "digital signature". The automatic CLA helps in identifying the statement and unblocking the merge if all contributors have signed the agreement. Isn't it good to have an instruction on how to sign it?

dietmarw replied:
OK I might have misunderstood the process how this cla-assistant works as opposed to the one we use in MSL https://cla-assistant.io where you simply tick a box.

dietmarw replied:
OK just saw the GIF and this part is fine.

dietmarw commented on .github/legal/cla-entity.md:
```diff
@@ -0,0 +1,85 @@
+
+Thank you for your interest in contributing to Open Instance Power System Library - OpenIPSL  ("We" or "Us").
+
+This contributor agreement ("Agreement") documents the rights granted by contributors to Us. To make this document effective, please sign it and send it to Us by electronic submission, following the instructions [here](README.md). This is a legally binding document, so please read it carefully before agreeing to it. The Agreement may cover more than one software project managed by Us.
```
```suggestion
This contributor agreement ("Agreement") documents the rights granted by contributors to Us. To make this document effective, please sign it and send it to Us by electronic submission, following the instructions given in the [README](../../README.md). This is a legally binding document, so please read it carefully before agreeing to it. The Agreement may cover more than one software project managed by Us.
```

dietmarw replied:
Better double check if the link is working. Relative links can be funny when looking at source location and deployment location.

marcelofcastro replied:
This is already correct. I was not mentioning OpenIPSL/README.md, I was mentioning .github/legal/README.md, which is the link for the CLA instructions.

dietmarw replied:
I see. OK I'll update the suggestion.

dietmarw commented on .github/legal/cla-individual.md:
```diff
@@ -0,0 +1,91 @@
+
+Thank you for your interest in contributing to Open Instance Power System Library - OpenIPSL  ("We" or "Us").
+
+This contributor agreement ("Agreement") documents the rights granted by contributors to Us. To make this document effective, please sign it and send it to Us by electronic submission, following the instructions [here](README.md). This is a legally binding document, so please read it carefully before agreeing to it. The Agreement may cover more than one software project managed by Us.
```
```suggestion
This contributor agreement ("Agreement") documents the rights granted by contributors to Us. To make this document effective, please sign it and send it to Us by electronic submission, following the instructions [README](../../README.md). This is a legally binding document, so please read it carefully before agreeing to it. The Agreement may cover more than one software project managed by Us.
```

dietmarw replied:
Better double check if the link is working. Relative links can be funny when looking at source location and deployment location.

marcelofcastro replied:
Same here.

dietmarw commented on .github/workflows/cla.yml:
```diff
@@ -0,0 +1,36 @@
+          PERSONAL_ACCESS_TOKEN : ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+        with:
+          path-to-signatures: '.github/legal/signatures/version1/cla.json'
+          path-to-document: '.github/legal/cla-individual.md' 
```
I'm pretty sure the documentation said that `path-to-document` needs to be an absolute URL.

marcelofcastro replied:
I can check that, but then I'd have to put the "future path", corresponding to the one we would have after this PR is merged. Right?

dietmarw replied:
Yes. So it would be: `https://github.com/OpenIPSL/OpenIPSL/blob/master/.github/legal/cla-individual.md`

dietmarw commented on .github/workflows/cla.yml:
```diff
@@ -0,0 +1,36 @@
+          #custom-pr-sign-comment: 'The signature to be committed in order to sign the CLA'
+          #custom-allsigned-prcomment: 'pull request comment when all contributors has signed, defaults to **CLA Assistant Lite bot** All Contributors have signed the CLA.'
+          #lock-pullrequest-aftermerge: false - if you don't want this bot to automatically lock the pull request after merging (default - true)
+          #use-dco-flag: true - If you are using DCO instead of CLA
```
```suggestion
          #use-dco-flag: true - If you are using DCO instead of CLA
          
```

marcelofcastro replied:
I can't see the changes in this suggestion other than a space being added. Is that a good practice?

dietmarw commented on .gitignore:
```diff
@@ -16,3 +16,4 @@ dymosim
 *.bak-mo
 *.moe
 *.mol
+.DS_Store
```
```suggestion
.DS_Store

```

marcelofcastro replied:
Same thing for the new row being added.

dietmarw replied:
See https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline GitHub even marks the missing newline with a red symbol

dietmarw commented on README.md:
```diff
@@ -4,15 +4,15 @@
 
 ## Scope
-The purposes of OpenIPSL are to be used for research and education (therefore frequent release times may be available), with maximum compatibility with [OpenModelica](http://openmodelica.org/) (to provide a free/libre and cost-free alternative for power system dynamic simulation), to provide as many as possible typical "test networks" for use in research and teaching, and to be developed in such  ...
+The OpenIPSL is developed to be used for research and education (therefore frequent release times may be available), with maximum compatibility with [OpenModelica](http://openmodelica.org/) (to provide a free/libre and cost-free alternative for power system dynamic simulation), to provide as many as possible typical "test networks" for use in research and teaching, and to be developed in such way ...
```
```suggestion
The OpenIPSL is developed to be used for research and education (therefore frequent release times may be available), with maximum compatibility with [OpenModelica](https://openmodelica.org/) (to provide a free/libre and cost-free alternative for power system dynamic simulation), to provide as many as possible typical "test networks" for use in research and teaching, and to be developed in such way that the library can efficiently be used for power system simulation within Modelica-based workflows (i.e., helping to give reference power system models for development and testing of Modelica back-end compilers) when faced with power system simulation challenges.
```

dietmarw commented on README.md:
```diff
@@ -23,54 +23,31 @@ iPSL is part of the [iTesla Tool](https://github.com/itesla/ipst), and thus, it
+Contributions to the library are welcome and can be submitted in the form of pull requests to this repository. Please consult the [contribution guidelines](.github/CONTRIBUTING.md) for more information on how to contribute to the development of this library. Information about our current and past contributors can be found when the library is loaded into a Modelica-compliant software. The informat ...
 
-Copyright April 2016 - May 2017 [SmarTS Lab](http://openipsl.readthedocs.io/en/latest/community/community-home.html); Prof. [Luigi Vanfretti's](https://github.com/lvanfretti) research group formerly at KTH Royal Institute of Technology, Stockholm, Sweden, now [ALSETLab](https://github.com/ALSETLab) at Rensselaer Polytechnic Institute, Troy, NY, USA.
+If you want to submit your contributions to the OpenIPSL, note that we have a [Contributor's License Agreement (CLA)](.github/legal/README.md). Individual contributors should read, accept and virtually sign the [Individual Contributor License Agreement](.github/legal/cla-individual.md) for contributions to be included in the OpenIPSL. Contributions submitted by entities need to have the [Entity C ...
```
I would simply link to the CLA guide you have created which also explains the automatic procedure and simply remove the rest of the paragraph.

dietmarw commented on README.md:
```diff
@@ -23,54 +23,31 @@ iPSL is part of the [iTesla Tool](https://github.com/itesla/ipst), and thus, it
 ## Citing OpenIPSL in Publications:
 If you use OpenIPSL in your work or research, all we ask you in exchange is that you **cite the reference publications**, according to your use.
-Please consult our [Publication List](http://openipsl.readthedocs.io/en/latest/publications.html), for browsing the reference publications.
+Please consult our publication list, located within the Users' Guide package, in the Publications page, for browsing the reference publications.
```
I think this whole section can be replaced by use of [CITATION files](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files).

marcelofcastro replied:
I checked their documentation and I believe that if we use CITATION.cff, it will not be able to create a citation for the PAPER itself, but the REPO. You see, for example, that the information is translated into @software{} in BibTeX and I am not sure that it is a good idea.

marcelofcastro replied:
Maybe @lvanfretti can also give his input in this matter.

marcelofcastro replied:
WAIT - I think we can configure it. Sorry. Let me check it first.

marcelofcastro replied:
Okay, I have created a cff file...

dietmarw reviewed (changes requested):
Nice work. As usual I'd like to have some small things polished. See my comments. You might have noticed that changed (hopefully) all of the "here" links [for a reason](https://uxmovement.com/content/why-your-links-should-never-say-click-here/).

marcelofcastro commented:
I think that I have addressed all of your comments, @dietmarw. Please take another look :)

dietmarw commented on README.md:
```diff
@@ -28,20 +28,15 @@ Documentation is provided within the library and can be accessed when loading Op
-Preferably, please cite the following paper:
-  - Maxime Baudette, Marcelo Castro, Tin Rabuzin, Jan Lavenius, Tetiana Bogodorova, Luigi Vanfretti, OpenIPSL: Open-Instance Power System Library — Update 1.5 to “iTesla Power Systems Library (iPSL): A Modelica library for phasor time-domain simulations”, SoftwareX, Volume 7, January–June 2018, Pages 34-36, ISSN 2352-7110, https://doi.org/10.1016/j.softx.2018.01.002
-    - Link to Publisher's Website: [SoftwareX](https://www.sciencedirect.com/science/article/pii/S2352711018300050)
+If you use OpenIPSL in your work or research, all we ask you in exchange is that you **cite the reference publications**, according to your use. Please consult our publication list, located within the Users' Guide package, in the Publications page, for browsing the reference publications. Preferrably, please cite this repository by using our preferred reference, as seen on GitHub GUI.
```
```suggestion
If you use OpenIPSL in your work or research, all we ask you in exchange is that you **cite the reference publications**, according to your use. Please consult our publication list, located within the User's Guide package, in the Publications page, for browsing the reference publications. Preferrably, please cite this repository by using our preferred reference, as seen on GitHub GUI.
```

dietmarw replied:
Looks good now with only this one typo.

marcelofcastro replied:
It was not a typo. I used Users' Guide because the guide supposedly belongs to all users and not to one specific user. Anyway, I think the singular will do the job since it is [an abstract representation of all users](https://english.stackexchange.com/questions/36657/user-s-guide-vs-users-guide#:~:text=The%20difference%20is%20in%20the,belonging%20to%20all%20the%20users.).
